obs-studio: Browser support broken

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

https://github.com/NixOS/nixpkgs/blob/1e5b653dff12029333a6546c11e108ede13052eb/pkgs/applications/video/obs-studio/default.nix#L41

Looks like the cef binary is old. obs-studio is working with a more recent version (in AUR it's `127.3.4`)

I don't know who attach the `browserSupport` as `fix me` but anyone take this. 


### Steps to reproduce

Instal obs-studio from unstable branch with browserSupport.

### Expected behaviour

Work with browserSupport

### Screenshots

_No response_

### Relevant log output

```console
error: builder for '/nix/store/54ha08lx0i2ly9b6vfyj2wm16vw11gd1-obs-studio-31.0.2.drv' failed with exit code 1;
       last 25 log lines:
       > /nix/store/gj9lra51hwhxnhz05jqk5lh03wipamv0-gcc-wrapper-14-20241116/bin/g++ -DBROWSER_AVAILABLE -DENABLE_HEVC -DHAVE_OBSCONFIG_H -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DSIMDE_ENABLE_OPENMP -Dobs_browser_EXPORTS -I/build/source/build/plugins/obs-browser -I/build/source/plugins/obs-browser -I/build/source/build/plugins/obs-browser/obs-browser_autogen/include -I/build/source/plugins/obs-browser/deps -I/build/source/build/config -I/build/source/libobs -I/build/source/UI/obs-frontend-api -I/build/source/plugins/obs-websocket/lib -I/build/source/plugins/obs-browser/panel -isystem /build/cef -isystem /nix/store/00dn0wh9afy98xja69p7i404jgznfkdk-qtbase-6.8.2/include/QtWidgets -isystem /nix/store/00dn0wh9afy98xja69p7i404jgznfkdk-qtbase-6.8.2/include/QtCore -isystem /nix/store/00dn0wh9afy98xja69p7i404jgznfkdk-qtbase-6.8.2/mkspecs/linux-g++ -isystem /nix/store/00dn0wh9afy98xja69p7i404jgznfkdk-qtbase-6.8.2/include/QtGui -O3 -DNDEBUG -std=c++17 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -fopenmp-simd -fno-strict-aliasing -Wdeprecated-declarations -Wempty-body -Wenum-conversion -Werror=return-type -Wextra -Wformat -Wformat-security -Wno-conversion -Wno-float-conversion -Wno-implicit-fallthrough -Wno-missing-braces -Wno-missing-field-initializers -Wno-shadow -Wno-sign-conversion -Wno-trigraphs -Wno-unknown-pragmas -Wno-unused-function -Wno-unused-label -Wparentheses -Wswitch -Wuninitialized -Wunreachable-code -Wunused-parameter -Wunused-value -Wunused-variable -Wvla -Winvalid-offsetof -Wno-overloaded-virtual -Wno-error=maybe-uninitialized -Winfinite-recursion -Wno-stringop-overflow -fPIC -Werror -MD -MT plugins/obs-browser/CMakeFiles/obs-browser.dir/browser-client.cpp.o -MF plugins/obs-browser/CMakeFiles/obs-browser.dir/browser-client.cpp.o.d -o plugins/obs-browser/CMakeFiles/obs-browser.dir/browser-client.cpp.o -c /build/source/plugins/obs-browser/browser-client.cpp
       > In file included from /build/source/plugins/obs-browser/browser-client.cpp:19:
       > /build/source/plugins/obs-browser/browser-client.hpp:101:22: error: 'virtual void BrowserClient::OnRenderProcessTerminated(CefRefPtr<CefBrowser>, CefRequestHandler::TerminationStatus, int, const CefString&)' marked 'override', but does not override
       >   101 |         virtual void OnRenderProcessTerminated(CefRefPtr<CefBrowser> browser, TerminationStatus status, int error_code,
       >       |                      ^~~~~~~~~~~~~~~~~~~~~~~~~
       > [253/534] Building C object plugins/obs-outputs/CMakeFiles/obs-outputs.dir/librtmp/rtmp.c.o
       > In function 'DecodeTEA',
       >     inlined from 'HandleInvoke' at /build/source/plugins/obs-outputs/librtmp/rtmp.c:3164:21:
       > /build/source/plugins/obs-outputs/librtmp/rtmp.c:4606:7: warning: '*(uint32_t *)out' may be used uninitialized [-Wmaybe-uninitialized]
       >  4606 |     y = v[0];
       >       |     ~~^~~~~~
       > [254/534] Building CXX object plugins/obs-browser/CMakeFiles/obs-browser.dir/panel/browser-panel.cpp.o
       > [255/534] Swig compile obspython.i for python
       > [256/534] Building CXX object plugins/obs-browser/CMakeFiles/obs-browser.dir/obs-browser-source.cpp.o
       > FAILED: plugins/obs-browser/CMakeFiles/obs-browser.dir/obs-browser-source.cpp.o
       > /nix/store/gj9lra51hwhxnhz05jqk5lh03wipamv0-gcc-wrapper-14-20241116/bin/g++ -DBROWSER_AVAILABLE -DENABLE_HEVC -DHAVE_OBSCONFIG_H -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DSIMDE_ENABLE_OPENMP -Dobs_browser_EXPORTS -I/build/source/build/plugins/obs-browser -I/build/source/plugins/obs-browser -I/build/source/build/plugins/obs-browser/obs-browser_autogen/include -I/build/source/plugins/obs-browser/deps -I/build/source/build/config -I/build/source/libobs -I/build/source/UI/obs-frontend-api -I/build/source/plugins/obs-websocket/lib -I/build/source/plugins/obs-browser/panel -isystem /build/cef -isystem /nix/store/00dn0wh9afy98xja69p7i404jgznfkdk-qtbase-6.8.2/include/QtWidgets -isystem /nix/store/00dn0wh9afy98xja69p7i404jgznfkdk-qtbase-6.8.2/include/QtCore -isystem /nix/store/00dn0wh9afy98xja69p7i404jgznfkdk-qtbase-6.8.2/mkspecs/linux-g++ -isystem /nix/store/00dn0wh9afy98xja69p7i404jgznfkdk-qtbase-6.8.2/include/QtGui -O3 -DNDEBUG -std=c++17 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -fopenmp-simd -fno-strict-aliasing -Wdeprecated-declarations -Wempty-body -Wenum-conversion -Werror=return-type -Wextra -Wformat -Wformat-security -Wno-conversion -Wno-float-conversion -Wno-implicit-fallthrough -Wno-missing-braces -Wno-missing-field-initializers -Wno-shadow -Wno-sign-conversion -Wno-trigraphs -Wno-unknown-pragmas -Wno-unused-function -Wno-unused-label -Wparentheses -Wswitch -Wuninitialized -Wunreachable-code -Wunused-parameter -Wunused-value -Wunused-variable -Wvla -Winvalid-offsetof -Wno-overloaded-virtual -Wno-error=maybe-uninitialized -Winfinite-recursion -Wno-stringop-overflow -fPIC -Werror -MD -MT plugins/obs-browser/CMakeFiles/obs-browser.dir/obs-browser-source.cpp.o -MF plugins/obs-browser/CMakeFiles/obs-browser.dir/obs-browser-source.cpp.o.d -o plugins/obs-browser/CMakeFiles/obs-browser.dir/obs-browser-source.cpp.o -c /build/source/plugins/obs-browser/obs-browser-source.cpp
       > In file included from /build/source/plugins/obs-browser/obs-browser-source.cpp:20:
       > /build/source/plugins/obs-browser/browser-client.hpp:101:22: error: 'virtual void BrowserClient::OnRenderProcessTerminated(CefRefPtr<CefBrowser>, CefRequestHandler::TerminationStatus, int, const CefString&)' marked 'override', but does not override
       >   101 |         virtual void OnRenderProcessTerminated(CefRefPtr<CefBrowser> browser, TerminationStatus status, int error_code,
       >       |                      ^~~~~~~~~~~~~~~~~~~~~~~~~
       > [257/534] Building CXX object plugins/obs-browser/CMakeFiles/obs-browser.dir/panel/browser-panel-client.cpp.o
       > [258/534] Building C object plugins/obs-outputs/CMakeFiles/obs-outputs.dir/mp4-mux.c.o
       > [259/534] Building CXX object plugins/obs-browser/CMakeFiles/browser-helper.dir/browser-app.cpp.o
       > [260/534] Building CXX object plugins/obs-browser/CMakeFiles/obs-browser.dir/browser-app.cpp.o
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/54ha08lx0i2ly9b6vfyj2wm16vw11gd1-obs-studio-31.0.2.drv'.
error: 1 dependencies of derivation '/nix/store/x4qs1bp2k98kv6j3qgrippqgx0l8f97z-obs-studio.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vma6gp9ggs9yswnqcd4wv96sjz6hzvnw-wrapped-obs-studio.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ffvi9xr2h75mdp5p54160flayxqpgh1k-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9p365m31p0hfwyls8f5b96fpiipv8l95-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mf3fqqzn1ifxmxayi8krvdsfqvmxfvfv-user-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2vwb46jynk1bjgwlrx1zirzz8948aax3-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/08z8clw89pqrnw9lxw92x0cqlij32lg5-nixos-system-aesthetic-25.05.20250319.a84ebe2.drv' failed to build
┏━ 1 Errors:
 ⋮
┃        > /nix/store/gj9lra51hwhxnhz05jqk5lh03wipamv0-gcc-wrapper-14-20241116/bin/g++ -DBROWSER_AVAILABLE -DENABLE_HEVC -DHAVE_OBSCONFIG_H -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT…
┃        > In file included from /build/source/plugins/obs-browser/obs-browser-source.cpp:20:
┃        > /build/source/plugins/obs-browser/browser-client.hpp:101:22: error: 'virtual void BrowserClient::OnRenderProcessTerminated(CefRefPtr<CefBrowser>, CefRequestHandler::Terminati…
┃        >   101 |         virtual void OnRenderProcessTerminated(CefRefPtr<CefBrowser> browser, TerminationStatus status, int error_code,
┃        >       |                      ^~~~~~~~~~~~~~~~~~~~~~~~~
┃        > [257/534] Building CXX object plugins/obs-browser/CMakeFiles/obs-browser.dir/panel/browser-panel-client.cpp.o
┃        > [258/534] Building C object plugins/obs-outputs/CMakeFiles/obs-outputs.dir/mp4-mux.c.o
┃        > [259/534] Building CXX object plugins/obs-browser/CMakeFiles/browser-helper.dir/browser-app.cpp.o
┃        > [260/534] Building CXX object plugins/obs-browser/CMakeFiles/obs-browser.dir/browser-app.cpp.o
┃        > ninja: build stopped: subcommand failed.
┃        For full logs, run 'nix log /nix/store/54ha08lx0i2ly9b6vfyj2wm16vw11gd1-obs-studio-31.0.2.drv'.
┣━ Dependency Graph:
┃                            ┌─ ⚠ obs-studio-31.0.2 failed with exit code 1 after ⏱ 56s in buildPhase
```

### Additional context

_No response_

### System metadata

```nu
 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.7-cachyos, NixOS, 25.05 (Warbler), 25.05.20250319.a84ebe2`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/linuxmobile/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/linuxmobile/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/linuxmobile/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/linuxmobile/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/riqr33jgc60nsk8w2d0blq17y8skp0yn-lix-2.91.1/share`
 - nixpkgs: `/nix/store/1728d3jg85mkz2w2cvk6vi74i30fn6x7-source`
```

### Notify maintainers

@jb55 @fpletz @materusPL 

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

--- a/pkgs/applications/video/obs-studio/default.nix
+++ b/pkgs/applications/video/obs-studio/default.nix
@@ -6,6 +6,7 @@
   ninja,
   nv-codec-headers-12,
   fetchFromGitHub,
+  fetchpatch,
   addDriverRunpath,
   autoAddDriverRunpath,
   cudaSupport ? config.cudaSupport,
@@ -38,7 +39,7 @@
   alsa-lib,
   pulseaudioSupport ? config.pulseaudio or stdenv.hostPlatform.isLinux,
   libpulseaudio,
-  browserSupport ? false, # FIXME: broken
+  browserSupport ? true,
   libcef,
   pciutils,
   pipewireSupport ? stdenv.hostPlatform.isLinux,
@@ -84,6 +85,33 @@ stdenv.mkDerivation (finalAttrs: {
     # Lets obs-browser build against CEF 90.1.0+
     ./Enable-file-access-and-universal-access-for-file-URL.patch
     ./fix-nix-plugin-path.patch
+    # TODO: remove when CHROME_VERSION_BUILD(libcef) >= 6367
+    (fetchpatch {
+      name = "Check-source-validity-before-attempting-to-log-rende.patch";
+      url = "https://github.com/obsproject/obs-browser/pull/478.patch";
+      revert = true;
+      stripLen = 1;
+      extraPrefix = "plugins/obs-browser/";
+      hash = "sha256-mQVhK4r8LlK2F9/jlDHA1V6M29mAfxWAU/VsMXYNrhU=";
+    })
+    # TODO: remove when CHROME_VERSION_BUILD(libcef) >= 6367
+    (fetchpatch {
+      name = "Print-browser-source-renderer-crashes-to-OBS-log.patch";
+      url = "https://github.com/obsproject/obs-browser/pull/475.patch";
+      revert = true;
+      stripLen = 1;
+      extraPrefix = "plugins/obs-browser/";
+      hash = "sha256-ha77OYpWn57JovPNE+izyDOB/2KlF3qWVv/PGEgyu84=";
+    })
+    # TODO: remove when CHROME_VERSION_BUILD(libcef) >= 6367
+    (fetchpatch {
+      name = "Log-error-if-CefInitialize-fails.patch.patch";
+      url = "https://github.com/obsproject/obs-browser/pull/477.patch";
+      revert = true;
+      stripLen = 1;
+      extraPrefix = "plugins/obs-browser/";
+      hash = "sha256-MMLFQtpWjfpti/38qEcOuXr1L3s1MPRHjuaZCjNmvt0=";
+    })
   ];
 
   nativeBuildInputs =

